lib/jekyll/filters/url_filters.rb

```diff
@@ -51,15 +56,21 @@ def strip_index(input)
       private
 
       def sanitized_baseurl
-        site = @context.registers[:site]
-        site.config["baseurl"].to_s.chomp("/")
+        @sanitized_baseurl ||= begin
+          ensure_leading_slash(
+            @context.registers[:site].config["baseurl"].to_s.chomp("/")
+          )
+        end
       end
 
       def ensure_leading_slash(input)
         return input if input.nil? || input.empty? || input.start_with?("/")
         "/#{input}"
       end
 
+      def absolute?(input)
+        %r!\A[a-zA-Z+.-]+:/!.match(input)
```

Comment:
Let's store this regexp as a constant so it's not re-created on each function call. If we can get away with no regexp, then all the better!